sweep/sweeper.go

```diff
@@ -413,29 +455,31 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch,
 			// be started when new inputs arrive.
 			s.timer = nil
 
-			// Retrieve fee estimate for input filtering and final
-			// tx fee calculation.
-			satPerKW, err := s.cfg.FeeEstimator.EstimateFeePerKW(
-				s.cfg.SweepTxConfTarget,
-			)
-			if err != nil {
-				log.Errorf("estimate fee: %v", err)
-				continue
-			}
-
-			// Examine pending inputs and try to construct lists of
-			// inputs.
-			inputLists, err := s.getInputLists(bestHeight, satPerKW)
-			if err != nil {
-				log.Errorf("get input lists: %v", err)
-				continue
-			}
-
-			// Sweep selected inputs.
-			for _, inputs := range inputLists {
-				err := s.sweep(inputs, satPerKW, bestHeight)
+			// We'll attempt to batch all of our inputs by their fee
+			// preference.
+			for _, bucket := range s.bucketManager.buckets {
+				// Examine pending inputs and try to construct
+				// lists of inputs.
+				feeRate := bucket.feeRate()
+				inputLists, err := s.getInputLists(
+					bucket.inputs, feeRate, bestHeight,
+				)
 				if err != nil {
-					log.Errorf("sweep: %v", err)
+					log.Errorf("Unable to examine pending "+
+						"inputs: %v", err)
+					continue
+				}
+
+				// Sweep selected inputs.
+				for _, inputs := range inputLists {
+					err := s.sweep(
+						inputs, feeRate,
+						bucket.sweepScript, bestHeight,
```

Comment:
Where is `sweepScript` ever set for a bucket?

Reply:
It'll be set within `sweep()` at the time of sweeping.

Reply:
I think we don't need a sweep script per bucket, but can just track the last unused address sweeper-wide.